packages/core/EventManager.ts

```diff
@@ -535,78 +535,104 @@ export default class EventManager {
-        if (destination.credentialId) {
-          let credential = this.calendarCredentials.find((c) => c.id === destination.credentialId);
-          if (!credential) {
-            // Fetch credential from DB
-            const credentialFromDB = await prisma.credential.findUnique({
-              where: {
-                id: destination.credentialId,
-              },
-              select: credentialForCalendarServiceSelect,
-            });
-            if (credentialFromDB && credentialFromDB.appId) {
-              credential = {
-                id: credentialFromDB.id,
-                type: credentialFromDB.type,
-                key: credentialFromDB.key,
-                userId: credentialFromDB.userId,
-                teamId: credentialFromDB.teamId,
-                invalid: credentialFromDB.invalid,
-                appId: credentialFromDB.appId,
-                user: credentialFromDB.user,
-              };
+
+            if (attemptedDestinationCalendarIds.includes(potentialDestinationCalendar.externalId)) {
+              continue;
             }
+            credentialId = potentialDestinationCalendar.credentialId;
+            destinationCalendar = potentialDestinationCalendar;
+            break;
           }
-          if (credential) {
-            const createdEvent = await createEvent(credential, event, destination.externalId);
-            if (createdEvent) {
-              createdEvents.push(createdEvent);
+        }
+
+        if (!credentialId) {
+          // Keep this logic here for other events that need to fallback to other calendars
+          if (eventCreationAttempts === 0) {
+            const destinationCalendarCredentials = this.calendarCredentials.filter(
+              (c) => c.type === destinationCalendar.integration
+            );
+            // It might not be the first connected calendar as it seems that the order is not guaranteed to be ascending of credentialId.
+            const firstCalendarCredential = destinationCalendarCredentials[0] as
+              | (typeof destinationCalendarCredentials)[number]
+              | undefined;
+
+            if (!firstCalendarCredential) {
+              log.warn(
+                "No other credentials found of the same type as the destination calendar. Falling back to first connected calendar"
+              );
+              await fallbackToFirstConnectedCalendar();
+            } else {
+              log.warn(
+                "No credentialId found for destination calendar, falling back to first found calendar of same type as destination calendar",
+                safeStringify({
+                  destination: getPiiFreeDestinationCalendar(destinationCalendar),
+                  firstConnectedCalendar: getPiiFreeCredential(firstCalendarCredential),
+                })
+              );
+              createdEvents.push(await createEvent(firstCalendarCredential, event));
+              eventCreated = true;
+              break;
```

Comment:
Not a huge fan of this approach but keeping this in here to maintain what we currently have.